ui-manchette-with-spacetimechart/src/helpers.ts

```diff
@@ -0,0 +1,166 @@
+import type {
+  OperationalPointType,
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+} from '@osrd-project/ui-manchette/dist/types';
+import type {
+  OperationalPoint,
+  SpaceTimeChartProps,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import { BASE_OP_HEIGHT, MAX_TIME_WINDOW, MAX_ZOOM_X, MIN_ZOOM_X } from './consts';
+import { getHeightWithoutLastWaypoint, msToS } from './utils';
+
+export const calcOperationalPointsToDisplay = (
+  operationalPoints: OperationalPointType[],
+  height: number,
+  isProportional: boolean,
+  zoom: number
+): StyledOperationalPointType[] => {
+  if (!isProportional || operationalPoints.length === 0) {
+    // For non-proportional display, we always display all the operational points:
+    return operationalPoints.map((op) => ({ ...op, styles: {}, display: true }));
+  }
+
+  // For proportional display, we only display points that do not overlap with
+  // the last displayed point:
+  const result: StyledOperationalPointType[] = [
+    { ...operationalPoints[0], display: true, styles: {} },
+  ];
+  const from = operationalPoints[0].position;
+  const to = operationalPoints.at(-1)!.position;
+  const totalDistance = to - from;
+  const heightWithoutFinalOp = getHeightWithoutLastWaypoint(height);
+  let lastDisplayedOP = result[0];
+
+  // We iterate through all points, and only add them if they don't collide
+  // with the last visible point:
+  for (let i = 1; i < operationalPoints.length; i++) {
+    const op = operationalPoints[i];
+    const diff = op.position - lastDisplayedOP.position;
+    const display = (diff / totalDistance) * heightWithoutFinalOp * zoom >= BASE_OP_HEIGHT;
+    result.push({
+      ...op,
+      display,
+      styles: {},
+    });
+
+    if (display) lastDisplayedOP = result[i];
```

Comment:
I think it would be better to add the op to result only if we should display it. Anyway, we don't use the non-displayed op in the code after this point ?
```suggestion
    if (display) {
      result.push({
        ...op,
        display,
      });
      lastDisplayedOP = result[i];
    }
```

The Manchette should display all the waypoints that it receives, so you can remove the display prop in StyledOperationalPointType (maybe rename it StyledWaypoint ?) and remove the early return `if(display) return null` in the component `OperationalPoint` (line 8)

Reply:
We can delay this comment

Reply:
Nice thanks !